ComputerVision/measurements.py

```diff
@@ -0,0 +1,170 @@
+
+def download () :
+    message = ""
+    user_input = simpledialog.askstring(title="Download", prompt="Enter file name:")
+    while True :
+        if os.path.isfile(user_input + ".txt") or os.path.isfile(user_input + ".png"):
+            user_input = simpledialog.askstring(title="Download", prompt="File \"" + user_input + ".txt\' already exists, enter new file name:")
+        else :
+            try :
+                f = open(user_input + ".txt", "w")
+                break
+            except OSError:
+                user_input = simpledialog.askstring(title="Download", prompt="File \"" + user_input + ".txt\' is an invalid file name, enter new file name:")
+
+    f.write("ID#, Cell Width\n")
+    for allpoints in settings.cellsizeidarray :
+        message += str(allpoints[0]) + ", " +str(allpoints[1]) + "\n"
+    f.write(message)
+    cv2.imwrite(user_input + ".png", settings.images["imagetomeasure"])
+    f.close()
+    messagebox.showinfo("Success", "Successfully downloaded.")
+
+def undo () :
+    if len(settings.previousImgs) != 0 :
+        if settings.previousImgs[len(settings.previousImgs)-1][1] == "circle" :
+            settings.points.pop()
+        elif settings.previousImgs[len(settings.previousImgs)-1][1] == "line" :
+            settings.variables['cellid'] -= 1
+            settings.cellsizeidarray.pop()
+            settings.points[len(settings.points)-2][2] = True
+        settings.images["imagetomeasure"] = np.array(settings.previousImgs[len(settings.previousImgs)-1][0])
+        settings.previousImgs.pop()
+        settings.previousImgsWithoutIDs.pop()
+        change_image()
+    else :
+        messagebox.showerror( "Error", "There is nothing to undo.")
+
+def hide_ids () :
+    if settings.hideids :
+        settings.buttons['hideIDsButton'].configure(text="Hide IDs")
+    else :
+        settings.buttons['hideIDsButton'].configure(text="Unhide IDs")
+    settings.hideids = not settings.hideids
+    change_image()
+        
+def change_image () :
+    if settings.hideids :
+        img = Image.fromarray(settings.previousImgsWithoutIDs[len(settings.previousImgsWithoutIDs)-1][0])
+        img = ImageTk.PhotoImage(image = img)
+        settings.imagelabels["measurementimage"].configure(image=img)
+        settings.imagelabels["measurementimage"].image = img
+        settings.images["measurementimage"] = img
+    else :
+        img = Image.fromarray(settings.images["imagetomeasure"])
+        img = ImageTk.PhotoImage(image = img)
+        settings.imagelabels["measurementimage"].configure(image=img)
+        settings.imagelabels["measurementimage"].image = img
+        settings.images["measurementimage"] = img
+
+def calculate_length (x1,x2,y1,y2) :
+    x,y,z = settings.images["imagetomeasure"].shape
+    inpixels = math.sqrt(math.pow(x1-x2,2)+math.pow(y1-y2,2))
+    distance = (settings.imagelength * inpixels) / x
+    settings.cellsizeidarray.append([settings.variables['cellid'], distance])
+
+def calculate () :
```

Comment:
calculate what?

Reply:
Changed the name for both to be more obvious